cleanlab/datalab/issue_manager/outlier.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, ClassVar, Dict, Optional, Tuple, Union, cast
+
+from scipy.sparse import csr_matrix
+from scipy.stats import iqr
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+
+from cleanlab.datalab.issue_manager import IssueManager
+from cleanlab.outlier import OutOfDistribution, transform_distances_to_scores
+
+if TYPE_CHECKING:  # pragma: no cover
+    from sklearn.neighbors import NearestNeighbors
+    from cleanlab.datalab.datalab import Datalab
+
+
+class OutlierIssueManager(IssueManager):
+    """Manages issues related to out-of-distribution examples."""
+
+    description: ClassVar[
+        str
+    ] = """Examples that are very different from the rest of the dataset 
+    (i.e. potentially out-of-distribution or rare/anomalous instances).
+    """
+    issue_name: ClassVar[str] = "outlier"
+    verbosity_levels = {
+        0: [],
+        1: [],
+        2: ["average_ood_score"],
+        3: [],
+    }
+
+    def __init__(
+        self,
+        datalab: Datalab,
+        threshold: Optional[float] = None,
+        **kwargs,
+    ):
+        super().__init__(datalab)
+
+        ood_kwargs = kwargs.get("ood_kwargs", {})
+
+        valid_ood_params = OutOfDistribution.DEFAULT_PARAM_DICT.keys()
+        params = {
+            key: value
+            for key, value in ((k, kwargs.get(k, None)) for k in valid_ood_params)
+            if value is not None
+        }
+
+        if params:
+            ood_kwargs["params"] = params
+
+        self.ood: OutOfDistribution = OutOfDistribution(**ood_kwargs)
+        self.threshold = threshold
+        self._embeddings: Optional[np.ndarray] = None
+        self._metric: str = None  # type: ignore
+
+    def find_issues(
+        self,
+        features: Optional[npt.NDArray] = None,
+        pred_probs: Optional[np.ndarray] = None,
+        iqr_scale: float = 1.5,
```

Comment:
what do you think about calling this arg `threshold` and redefining it so lower values imply less issues detected. This could be done by defining it as 1 / iqr_scale, ie. 0.666 (= 1 / 1.5) by default, and requiring threshold to be in 0-1.

Reply:
also why is this arg in `find_issues()` but the analogous `threshold` arg in NearDuplicate issue manager is in the `init()` ?

I would also move this arg to the init. Then you could easily just save it as an internal attribute of this OutlierIssueManager and then utilize it in `find_issues()` as is currently done.

Reply:
This arg is only used if `threshold` is None, which is set in the `init()`. this was specifically introduced to test IQR-rule-based thresholds.

Reply:
ah ok let's keep while still testing but don't forget to remove it after!